api/src/application/user/user-service.js

```diff
@@ -30,49 +31,73 @@ module.exports = (bookshelf, emailService, renderTemplate) => {
       return model.fetchAll();
     },
 
-    sendInvite(user, appName, hoursTillExpiration) {
+    sendInvite(user, appName, clientId, redirect_uri, scope, hoursTillExpiration) {
       return self.createPasswordResetToken(user.get('id'), hoursTillExpiration).then(token => {
         const base = config('/baseUrl');
+        const url = encodeURI(`${base}/user/accept-invite?token=${token.get('token')}&client_id=${clientId}&redirect_uri=${redirect_uri}&scope=${scope}`);
         return renderTemplate('email/invite', {
-          url: `${base}/user/accept-invite?token=${token.get('token')}`,
+          url: url.replace(' ', '%20'),
           appName: appName
-        });
-      }).then(emailBody => {
-        return emailService.send({
-          to: user.get('email'),
-          subject: `${appName} Invitation`,
-          html: emailBody,
-        });
+        }).then(emailBody => {
+          return emailService.send({
+            to: user.get('email'),
+            subject: `${appName} Invitation`,
+            html: emailBody,
+          }).then(data => {
+            return data;
+          }).catch(error => {
+            Boom.badImplementation('Something went wrong', error);
+          });
+        }).catch(error => {
+          Boom.badImplementation('Something went wrong', error);
+        });  
+      }).catch(error => {
+        Boom.badImplementation('Something went wrong', error);
```

Comment:
I don't think these three `.catch()` calls are doing anything either, and I don't think you need them.

These `.catch()` calls create new Boom error objects but they don't `throw` them. I think the actual effect of these calls is that they will catch errors then create a new error object then discard that error object and fail silently.

If you just delete all these `.catch()` calls the framework will catch rejected promises for you and correctly return a `Boom.badImplementation()` object.